src/platform/mbed/ConfigurationManagerImpl.h

```diff
@@ -0,0 +1,103 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *          Provides an implementation of the ConfigurationManager object
+ *          for mbed platforms.
+ */
+
+#pragma once
+
+#include <platform/internal/GenericConfigurationManagerImpl.h>
+#include <platform/mbed/MbedConfig.h>
+
+// Forward declare test method that run the tests.
+int cmd_device_test_config(int argc, char ** argv);
```

Comment:
Why doesn't the command call `ConfigurationMgr().RunUnitTests` which is public and wouldn't need such "friend" declarations?

Reply:
Because we want this to be available even if `NDEBUG` is defined. That's not possible today: 
https://github.com/project-chip/connectedhomeip/blob/17be4afe670126e94282bcf14a57aa3984e36cf4/src/include/platform/ConfigurationManager.h#L570-L571 .